### DISCUSSION TITLE
Calculate cloud cover from cloud layers

### DISCUSSION THREAD

**Discussion by PeterGrant** [weather-gov/api#420] [Q&A]
Hello,

I have a script calling the API to collect current weather data from 16 sites around the country, and need to estimate current solar irradiation from the available dataset. There are several existing algorithms to calculate irradiation from cloud cover; however, that is not included in the data set I get from the weather.gov API. Does anybody know a way to calculate cloud cover or solar irradiation from the data available via the API?

For reference, here's a list of points I'm getting from the /stations/{stationId}/observations/latest command:

- temperature
- dewpoint
- windDirection
- windSpeed
- windGust
- barometricPressure
- seaLevelPressure
- visibility
- maxTemperatureLast24Hours
- minTemperatureLast24Hours
- precipitationLastHour
- precipitationLast3Hours
- precipitationLast6Hours
- relativeHumidity
- windChill
- heatIndex
- cloudLayers

The cloudLayers dictionary provides a dictionary of variable length with entries showing different cloud layer measurements in meters. 

Thanks,
-Peter

**Comment by sean-user** [comment]
You can get skyCover from the API which has a good correlation with the solar irradiance. I use it daily with my solar system to estimate production and monitor performance.

Try:  https://api.weather.gov/gridpoints/LOX/162,46 with gridpoint values for your location(s) and look for "skyCover" in the response.

**Reply by PeterGrant** [reply]
Thanks @sean-user! That was a very helpful tip.
